src/server/events/execution_event_bus.ts

```diff
@@ -1,28 +1,192 @@
-import { EventEmitter } from 'events';
-
 import { Message, Task, TaskStatusUpdateEvent, TaskArtifactUpdateEvent } from '../../types.js';
 
 export type AgentExecutionEvent = Message | Task | TaskStatusUpdateEvent | TaskArtifactUpdateEvent;
 
+/**
+ * Listener type for 'event' events that receive an AgentExecutionEvent payload.
+ */
+export type EventListener = (event: AgentExecutionEvent) => void;
+
+/**
+ * Listener type for 'finished' events that receive no payload.
+ */
+export type FinishedListener = () => void;
+
 export interface ExecutionEventBus {
   publish(event: AgentExecutionEvent): void;
-  on(eventName: 'event' | 'finished', listener: (event: AgentExecutionEvent) => void): this;
-  off(eventName: 'event' | 'finished', listener: (event: AgentExecutionEvent) => void): this;
-  once(eventName: 'event' | 'finished', listener: (event: AgentExecutionEvent) => void): this;
+  on(eventName: 'event', listener: EventListener): this;
+  on(eventName: 'finished', listener: FinishedListener): this;
+  off(eventName: 'event', listener: EventListener): this;
+  off(eventName: 'finished', listener: FinishedListener): this;
+  once(eventName: 'event', listener: EventListener): this;
+  once(eventName: 'finished', listener: FinishedListener): this;
```

Comment:
This is going to be at least build-time breaking change for users implemented this interface (or inherited from `DefaultExecutionEventBus`), as both are exported via `server/index.ts`.

I.e. rolling back changes your changes in a sample prevents it from compiling.

Let's try to stick with current public API so that we can benefit from this change before major version release (no exact plans at the moment).

`extensions.ts` can be used as a "compatibility test" and shouldn't require any changes.

Reply:
Good point. I'll go back and line things up with the existing API